### ISSUE TITLE
[java] UseStringBufferForStringAppends: False Positive with ternary operator

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#222]
**Rule Set:** java-optimizations/UseStringBufferForStringAppends

**Description:**
False Positive, similar to https://sourceforge.net/p/pmd/bugs/1340/, but with flipped condition and arguments.

**Code Sample demonstrating the issue:**

```
public class UseStringBuffer {
    public void foo() {
        String country = request.getParameter("country");
        country = (country != null) ? country : "USA";
    }
}
```

**Running PMD through:** CLI

**Comment by Oowekyala** [comment]
@adangel I've opened a PR with a fix (#307)
